build.gradle

```diff
@@ -77,68 +78,99 @@ project(':configurator') {
-    jreHashes = ['74daf0b77a7fd679cbb3a6228e0efa8c4a90b7664aa057f211e34bbfb38640fb',
-                 'c09bab89cd82483c371597c5c364094a145c1fbba43a1d3d7c3e350b89dedc89',
-                 '80855320c42a3b06617c6e466c64df67731542a805972185b075ca6cb1222c7f']
-    packageFileExtension = ['zip', 'tar.gz']
+
+final TARGZ = 'tar.gz', ZIP = 'zip', ZIPTASK = 'Zip', TARTASK = 'Tar', WINDOWS = 'windows', MAC = 'mac', LINUX = 'linux'
+final jreDownloadDir = file("$buildDir/jre-downloads")
+final jreExtractDir = "${jreDownloadDir.absolutePath}/jre"
+final packageDir = "${buildDir}/distributions/packages"
+final jarDir = "${buildDir}/libs"
+
+class PackageProperties {
+    String packageFileExt
+    String scriptFile
+    String taskType
+    boolean fileCompression
+}
+
+class JreProperties {
+    String jreHash
+    String jreFileExt
+    String jreFileName
+    String jreDownloadUrl
 }
 
+def packagePropertiesMap = [
+        windows: new PackageProperties(packageFileExt: ZIP, scriptFile: 'Windows-launcher.bat', taskType: ZIPTASK, fileCompression: false),
+        linux: new PackageProperties(packageFileExt: TARGZ, scriptFile: 'Linux-launcher.sh', taskType: TARTASK, fileCompression: true),
+        mac: new PackageProperties(packageFileExt: TARGZ, scriptFile: 'Mac-launcher.sh', taskType: TARTASK, fileCompression: true)
+]
+
+def jrePropertiesMap = [
+        windows: new JreProperties(jreFileExt: ZIP, jreHash: 'c09bab89cd82483c371597c5c364094a145c1fbba43a1d3d7c3e350b89dedc89'),
+        linux: new JreProperties(jreFileExt: TARGZ, jreHash: '74daf0b77a7fd679cbb3a6228e0efa8c4a90b7664aa057f211e34bbfb38640fb'),
+        mac: new JreProperties(jreFileExt: TARGZ, jreHash: '80855320c42a3b06617c6e466c64df67731542a805972185b075ca6cb1222c7f')
+]
+
 import java.security.MessageDigest
 
 task downloadJre(description: 'Download OpenJRE of Linux, Windows and Mac, and check their hashes') {
-
     doFirst {
         if(!jreDownloadDir.exists()) {
             jreDownloadDir.mkdirs()
         }
-        osNames.each {String sys ->
-            def index = osNames.indexOf(sys)
-            def jreFileExtension = (sys == 'windows' ? packageFileExtension[0] : packageFileExtension[1])
-            def jreFileName = jreFileNameTemplate(sys,jreFileExtension)
-            def jreFile = file("$jreDownloadDir.absolutePath/$jreFileName")
-            if(!jreFile.exists()){
-                new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/$jreFileName")
-                        .withInputStream {
-                    jreFile.newOutputStream() << it
-                }
-                final hash = MessageDigest.getInstance('SHA-256').with {
-                    jreFile.eachByte(1024) {buff,len ->
-                        update(buff, 0, len)
+        
+        println("Start downloading OpenJRE...")
+        jrePropertiesMap.collect {String sys,JreProperties props ->
+            Thread.start {
+                props.with {
```

Comment:
As mentioned, having a `.with` up this high is probably a bit overkill. Probably not too difficult either for the one or two calls you'd need to make with `props.` so let's remove that extra nesting.